About for the string type concept (#68)
exercism/cairo: concepts/strings/about.md

# String Types

## String Type

Cairo do not have a native data type for strings. However, there are two ways to form a string type: short strings and ByteArray.

### Short strings

A short string is formed with a single quote where each character that makes up the string is encoded on one byte - [see ASCII table](https://www.asciitable.com/).

As a ASCII string, for instance, 'abc' will be:

- 'a' results to 0x61
- 'b' results to 0x62
- 'c' results to 0x63
- 'abc' results to 0x616263

The felt252 data type can hold short strings type in Cairo. A felt252 short string is limited to 31 characters due to the size of felt252 being 251 bits.

Examples of short strings declarations:

```rust
       let my_first_char = 'D';
       let my_first_char_in_hex = 0x44;

       let my_first_string = 'Hello world';
       let my_first_string_in_hex = 0x48656C6C6F20776F726C64;
```

> ### ByteArray

Reviewer: @Ephraim-nonso pinging you in case you missed this change request

The ByteArray struct was introduced in Cairo 2.4.0. The ByteArray strings are written in double quotes as follows:

```Rust
    let long_string: ByteArray = "this is a string which has more than 31 characters";
```

> Technically, the update also introduced the `bytes31` which holds 31 bytes and which fits into a felt. The ByteArray struct is made of an array of 31 byte, a single felt and length for the remainder.

Reviewer: ```suggestion
Technically, the update also introduced the `bytes31` type which holds 31 bytes and which fits into a felt. The `ByteArray` struct basically represents long strings using an array of `bytes31` that hold "whole words" (i.e. words that fill the whole felt) and a single felt field that holds an "incomplete word" (the number of bytes within are specified in a separate length field).
```
I like that you investigated the `ByteArray` struct implementation, and described it, nice!

nit: just added a bit more context

# String Types

> Cairo does not have a native data type for strings. However, there are two ways to represent a string type: short strings using a single `felt252` and arbitrary length using `ByteArray`.

Reviewer: ```suggestion
Cairo does not have a native data type for strings. However, there are two ways to represent a string type: short strings using a single `felt252` and arbitrary length strings using `ByteArray`.
```

> Cairo does not have a native data type for strings. However, there are two ways to represent a string type: short strings using a single `felt252` and arbitrary length using `ByteArray`.

Reviewer: ```suggestion
Cairo does not have a native data type for strings. However, there are two ways to represent a string type: short strings using a single `felt252` and arbitrary length strings using `ByteArray`.
```
